GameArena.java

```diff
@@ -339,6 +339,8 @@ private void addThing(Object o, int layer)
 						this.things.add(position, o);
 						added = true;
 					}
+
+					position++;
 				}
```

Comment:
ok..... how much have you actually tested this patch?

Reply:
I was testing it when I came across this. Probably should've done before making the PR. My bad.